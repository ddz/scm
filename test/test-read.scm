;;; Some simple objects
#t
#T
#f
#F
#\space
#\newline
#\nEwLiNe
#\SpAcE
#\ 
#\1

;;; Numbers
1
#b10101
#d1234
#xbeef
#o666

;;; Symbols
abcd

;;; Lists
()
(a  b)
(a   .   b)
(a . (b . (c d)))
(a . (b . (c . d))) ; Improper list

;;; Vectors
#()
#(1 2 3 4)
#((1 2) (2 3) (3 4))
