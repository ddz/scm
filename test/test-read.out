> ;;; Some simple objects
> #t
#t
> #T
#t
> #f
#f
> #F
#f
> #\space
#\space
> #\newline
#\newline
> #\nEwLiNe
#\newline
> #\SpAcE
#\space
> #\ 
#\space
> #\1
#\1
> 
> ;;; Numbers
> 1
1
> #b10101
21
> #d1234
1234
> #xbeef
48879
> #o666
438
> 
> ;;; Symbols
> abcd
abcd
> 
> ;;; Lists
> ()
()
> (a  b)
(a b)
> (a   .   b)
(a . b)
> (a . (b . (c d)))
(a b c d)
> (a . (b . (c . d))) ; Improper list equivalent to (a b c . d)
(a b c . d)
> 
> ;;; Vectors
> #()
#()
> #(1 2 3 4)
#(1 2 3 4)
> #((1 2) (2 3) (3 4))
#((1 2) (2 3) (3 4))
> 